x-pack/plugins/cases/server/client/attachments/get.ts

```diff
@@ -132,9 +132,10 @@ export async function find(
   );
 
   if (
-    queryParams?.page &&
-    queryParams?.perPage &&
-    queryParams?.page * queryParams?.perPage > MAX_DOCS_PER_PAGE
+    typeof queryParams?.page === 'number' &&
+    typeof queryParams?.perPage === 'number' &&
+    (queryParams?.perPage > MAX_DOCS_PER_PAGE ||
+      queryParams?.page * queryParams?.perPage > MAX_DOCS_PER_PAGE)
```

Comment:
I think is better if we have a function to contain the logic of the validation. We also need to check if the page is also bigger than `MAX_DOCS_PER_PAGE`. What do you think if we do it like: 

```
const validateFindPagination = (params?: FindCommentsQueryParams) => {
  if (params?.page == null && params?.perPage == null) {
    return;
  }

  const pageAsNumber = params.page ?? 0;
  const perPageAsNumber = params.perPage ?? 0;

  const errorMsg =
    'The number of documents is too high. Paginating through more than 10,000 documents is not possible.';

  if (Math.max(pageAsNumber, perPageAsNumber) > MAX_DOCS_PER_PAGE) {
    throw Boom.badRequest(errorMsg);
  }

  if (pageAsNumber * perPageAsNumber > MAX_DOCS_PER_PAGE) {
    throw Boom.badRequest(errorMsg);
  }
};
```

Reply:
Also, we can add some unit tests to check all the combinations:

- page and perPage are undefined
- page is bigger than 10K
- perPage is bigger than 10K
- page * perPage is bigger than 10K

Reply:
done 🙌